### PR TITLE
Search-form-use-Gem-ransack

### PR DESCRIPTION
**Pull request by HoangVanTrinh92**
Add form search user use Gem Ransack

### CODE REVIEW

File: app/controllers/admin/users_controller.rb [hunk 1]
```diff
@@ -2,7 +2,8 @@ class Admin::UsersController < ApplicationController
   before_action :authenticate_user!
 
   def index
-    @users = User.all.paginate page: params[:page], per_page: Settings.paginate.normal
+    @search = User.search(params[:q])
```
Comment (dieunb) [inline app/controllers/admin/users_controller.rb]: remove `()`

File: app/helpers/users_helper.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  def display_results_header(count)
+    if count > results_limit
+      "Your first #{results_limit} results out of #{count} total"
+    else
+      "Your #{pluralize(count, 'result')}"
+    end
+  end
+
+  def action
+    action_name == 'advanced_search' ? :post : :get
```
Comment (dieunb) [inline app/helpers/users_helper.rb]: `"'advanced_search"`

File: app/helpers/users_helper.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+  def model_fields
+    [:id, :email]
+  end
+
+  def results_limit
+    10
+  end
+
+  def display_query_sql(users)
```
Comment (dieunb) [inline app/helpers/users_helper.rb]: remove `()`

File: app/helpers/users_helper.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    "SQL: #{users.to_sql}"
+  end
+
+  def display_results_header(count)
```
Comment (dieunb) [inline app/helpers/users_helper.rb]: remove `()`

File: app/helpers/users_helper.rb [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    else
+      "Your #{pluralize(count, 'result')}"
+    end
+  end
+
+  def action
+    action_name == 'advanced_search' ? :post : :get
+  end
+
+  def display_sort_column_headers(search)
```
Comment (dieunb) [inline app/helpers/users_helper.rb]: remove `()`

File: app/helpers/users_helper.rb [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    model_fields.each_with_object('') do |field, string|
+      string << content_tag(:th, sort_link(search, field, {}, method: action))
+    end
+  end
+
+  def display_search_results(objects)
```
Comment (dieunb) [inline app/helpers/users_helper.rb]: remove `()`

File: app/helpers/users_helper.rb [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+    objects.limit(results_limit).each_with_object('') do |object, string|
+      string << content_tag(:tr, display_search_results_row(object))
+    end
+  end
+
+  def display_search_results_row(object)
```
Comment (dieunb) [inline app/helpers/users_helper.rb]: remove `()`

File: app/helpers/users_helper.rb [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+    model_fields.each_with_object('') do |field, string|
+      string << content_tag(:td, object.send(field))
+    end
```
Comment (dieunb) [inline app/helpers/users_helper.rb]: `end.html_safe`

File: config/routes.rb [hunk 9]
```diff
@@ -9,6 +9,7 @@
     resources :users do
       resource :position_users, only: :show
       get "positions" => "position_users#show"
+      collection { post :search, to: "users#index" }
```
Comment (dieunb) [inline config/routes.rb]: `{post :search, to: "users#index"}`

File: app/views/admin/users/index.html.erb [hunk 10]
```diff
@@ -1,4 +1,15 @@
-<div id="users-list">     
-  <%= render partial: "user", collection: @users %>
-  <%= will_paginate %>
-</div>
+<h1>Search Users</h1>
```
Comment (khanhkhanh) [inline app/views/admin/users/index.html.erb]: i18n

File: app/views/admin/users/index.html.erb [hunk 11]
```diff
@@ -1,4 +1,15 @@
...
+<%= search_form_for @search, url: admin_users_path do |f| %>
+  <fieldset>
+    <legend>User</legend>
```
Comment (khanhkhanh) [inline app/views/admin/users/index.html.erb]: i18n

File: app/helpers/users_helper.rb [hunk 12]
```diff
@@ -0,0 +1,45 @@
+module UsersHelper
+
```
Comment (khanhkhanh) [inline app/helpers/users_helper.rb]: Remove this  line

**Pull request merged**